packages/mui-utils/src/resolveProps/resolveProps.ts

```diff
@@ -40,10 +45,21 @@ export default function resolveProps<
               (output[propName] as Record<string, unknown>)[slotPropName] = resolveProps(
                 (defaultSlotProps as Record<string, any>)[slotPropName],
                 (slotProps as Record<string, any>)[slotPropName],
+                mergeClassNameAndStyle,
               );
             }
           }
         }
+      } else if (propName === 'className' && mergeClassNameAndStyle) {
+        output[propName] = clsx(
+          defaultProps?.[propName] as string,
+          props?.[propName] as string,
```

Comment:
I think this can be explicit.
```suggestion
        output.className = clsx(
          defaultProps.className,
          props.className,
```